### PR TITLE
Refactored nepali number

### PR DESCRIPTION
**Pull request by aj3sh**
- Changed the method names, made more readable.
- Refactored functionalities of `number`. 
- Depreciated old methods.
- Added black's check only on pre-commit hook.

### CODE REVIEW

File: nepali/number.py [hunk 1]
```diff
@@ -1,58 +1,111 @@
...
+
+def english_to_nepali(number: Any) -> str:
+    """
+    Converts english number to nepali.
+    """
+    number = str(number)
+    converted_number = []
+    for n in number:
+        if n.isdigit() and int(n) in range(0, 10):
+            converted_number.append(str(NP_NUMBERS[int(n)]))
```
Comment (sugat009) [inline nepali/number.py]: NP_NUMBERS[int(n)] should return a str value and this conversion might not be necessary?

File: nepali/number.py [hunk 2]
```diff
@@ -1,58 +1,111 @@
...
+    return "".join(converted_number)
+
+
+def nepali_to_english(number: Any) -> str:
+    """
+    Converts nepali number to english.
+    """
+    number = str(number)
+    converted_number = []
+    nepali_number_set = set(NP_NUMBERS)
```
Comment (sugat009) [inline nepali/number.py:28]: not really necessary?

File: nepali/number.py [hunk 3]
```diff
@@ -1,58 +1,111 @@
...
+
+def add_comma_english(number: Any) -> str:
+    """
+    Adds comma in english style
+    Eg. 123456789 => 123,456,789
+    """
+    return "{:,}".format(int(number))
+
+
+def add_comma(number: Any, convert=False) -> str:
```
Comment (sugat009) [inline nepali/number.py:45]: convert unused

**Review by sugat009 — CHANGES_REQUESTED** [review]

File: nepali/number.py (hunk 2 above)
Comment (aj3sh) [reply nepali/number.py:28]: used for avoiding the O(n^2) loop below.

Comment (sugat009) [reply nepali/number.py:28]: `if n in nepali_number_set:`
i think this can be done with `NP_NUMBERS` itself

**Review by sugat009 — CHANGES_REQUESTED** [review]

File: nepali/number.py (hunk 2 above)
Comment (aj3sh) [reply nepali/number.py:28]: `n in nepali_number_set` is used inside a loop and using a set gives O(1) time complexity. If we use `list` here it becomes O(N).

File: nepali/number.py (hunk 1 above)
Comment (aj3sh) [reply nepali/number.py]: Updated

File: nepali/number.py (hunk 3 above)
Comment (aj3sh) [reply nepali/number.py:45]: Updated

**Review by sugat009 — APPROVED** [review]

**Pull request merged**